### PR TITLE
Remove partitioned reward test CLI and Config

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

Partitioned epoch rewards feature has been activated. The test cli and config are no longer needed. Clean it up to make the code simpler.

#### Summary of Changes
- remove partitioned reward test code

Fixes #

### CODE REVIEW

**Comment by brooksprumo** [comment]
IMO it would be nice to wait until 2.3 before removing the featurization code. Otherwise, if a private cluster is running and hasn't activated P.E.R. yet, when a node upgrades to master, it'll diverge. By waiting one additional version, then we know builds of edge (2.2) always work with builds of stable (2.0).


For this PR, is this *strictly* the testing/experimental code though?

**Comment by HaoranYi** [comment]
This pr didn't remove featurization code  it only removes test harness code and test cli and a few hacks.

**Comment by HaoranYi** [comment]
For this PR, is this strictly the testing/experimental code though?

Yes. 
I think the cli maybe obsolete. If we run the validator with the cli now on mainnet,  we may get hash mismatch because of the hack we put in to explicitly exclude rewards sysvar ...
I will double check...

**Comment by CriesofCarrots** [comment]
> IMO it would be nice to wait until 2.3 before removing the featurization code. Otherwise, if a private cluster is running and hasn't activated P.E.R. yet, when a node upgrades to master, it'll diverge. By waiting one additional version, then we know builds of edge (2.2) always work with builds of stable (2.0).
> 
> For this PR, is this _strictly_ the testing/experimental code though?

@brooksprumo Is this a new policy? cc @willhickey 
I believe the [documented process](https://github.com/anza-xyz/agave/wiki/Feature-Gate-Activation-Process) is to remove in master when the last level (MNB) has been activated, so remove in edge when activation happened on stable. (I'm asking because I've been working on removing the featurization code: https://github.com/CriesofCarrots/solana/tree/cleanup-per)

I think adding an additional minor-version delay is arbitrary. Consider the situation when v2.1 becomes stable and we've removed the PER feature in v2.3 (edge). A private cluster running v2.1 but without PER activated will experience the same divergence updating to master. That is to say, there isn't any way to *ever* remove featurization code and guarantee builds of edge always work with builds of stable. In every case, we must depend on users to emulate MNB activation sets.

**Comment by willhickey** [comment]
> Is this a new policy?

I think the policy has just been compatibility between adjacent minor versions, so removing v2.0 feature gated code in master is fine. But teams could choose to do cleanup slower than that, if they want

Edit: Fixed confusing formatting

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -5676,16 +5661,13 @@ impl Bank {
         let measure_total = Measure::start("");
 
         let slot = self.slot();
-        let ignore = (!self.is_partitioned_rewards_feature_enabled()
```
Comment (HaoranYi) [inline runtime/src/bank.rs:5679]: It looks like the test CLI will work fine even when the feature is active.

**Comment by HaoranYi** [comment]
> For this PR, is this strictly the testing/experimental code though?
> 
> ~Yes. I think the cli maybe obsolete. If we run the validator with the cli now on mainnet, we may get hash mismatch because of the hack we put in to explicitly exclude rewards sysvar ...~ I will double check...

ah. It will be fine to run validator with those test CLI.

**Comment by HaoranYi** [comment]
It looks like @CriesofCarrots's `cleanup_pr` branch already covered the changes in this PR. 

Let's close this one.

**Comment by brooksprumo** [comment]
> @brooksprumo Is this a new policy? 

and 

> I think adding an additional minor-version delay is arbitrary.

No, not a new policy, and yes, it is arbitrary.

Mostly I got bit by this on a pop-net cluster when TVC activated. I upgraded one node that was running the same minor version (2.2/master in this case), and then it stopped working because TVC's featurization was removed promptly after it was activated on mnb.

**Comment by CriesofCarrots** [comment]
> It looks like @CriesofCarrots's `cleanup_pr` branch already covered the changes in this PR.
> 
> Let's close this one.

Actually, I hadn't done any of the CLI or accountsDB cleanup, so I was going to approve this. Okay if I cherry-pick your commit instead, @HaoranYi ?

**Comment by HaoranYi** [comment]
> > It looks like @CriesofCarrots's `cleanup_pr` branch already covered the changes in this PR.
> > Let's close this one.
> 
> Actually, I hadn't done any of the CLI or accountsDB cleanup, so I was going to approve this. Okay if I cherry-pick your commit instead, @HaoranYi ?

Yes, please cherry pick it to your branch.

**Pull request closed without merging**